packages/client/src/DataCommonsClient.ts

```diff
@@ -0,0 +1,480 @@
+          const geometry = JSON.parse(geometryString) as Geometry;
+          const properties = { ...dataRow };
+          delete properties[`entity.${geoJsonProperty}`];
+          const feature: Feature = {
+            type: "Feature",
+            properties,
+            geometry,
+          };
+          return feature;
+        }),
+    };
+    return geoJson;
+  }
+
+  /**
+   * Fetches data commons variable values about an entity or entities.
+   * @param params
+   * @returns
+   */
+  async getDataRows(params: GetDataRowsParams): Promise<DataRow[]> {
+    // Fetch variable observations
+    const pointApiResponse =
+      "parentEntity" in params
+        ? await this.getObservationsPointWithin(params)
+        : await this.getObservationsPoint(params);
+    if (!pointApiResponse) {
+      return [];
+    }
+    const entityDcids =
+      this.getEntityDcidsFromPointApiResponse(pointApiResponse);
+    // Fetch relevant entity and variable property values
+    const entityPropValues = await this.getNodePropValues(
+      entityDcids,
+      params.entityProps || DEFAULT_ENTITY_PROPS
+    );
+    const variablePropValues = await this.getNodePropValues(
+      params.variables,
+      params.variableProps || DEFAULT_VARIABLE_PROPS
+    );
+
+    // Fetch population data for per capita calculations
+    let populationObservations: SeriesApiResponse = { data: {}, facets: {} };
+    if (!_.isEmpty(params.perCapitaVariables)) {
+      populationObservations =
+        "parentEntity" in params
+          ? await this.getObservationsSeriesWithin({
+              ...params,
+              variables: [TOTAL_POPULATION_VARIABLE],
+            })
+          : await this.getObservationsSeries({
+              ...params,
+              variables: [TOTAL_POPULATION_VARIABLE],
+            });
+    }
+
+    const dataRows = this.getDataRowsFromObservations(
+      entityDcids,
+      params.variables,
+      pointApiResponse,
+      entityPropValues,
+      variablePropValues,
+      populationObservations
+    );
+
+    return Promise.resolve(dataRows);
+  }
+
+  async getFirstNodeValues(params: {
+    dcids: string[];
+    prop: string;
+  }): Promise<Record<string, string | null>> {
+    const nodePropvals = await this.getNodePropvals(params);
+    const nodeValues: Record<string, string | null> = {};
+    Object.keys(nodePropvals).forEach((variableDcid) => {
+      nodeValues[variableDcid] =
+        nodePropvals[variableDcid].length > 0
+          ? nodePropvals[variableDcid][0].value
+          : null;
+    });
+    return nodeValues;
+  }
+
+  async getNodePropvals(params: {
+    dcids: string[];
+    prop: string;
+  }): Promise<ApiNodePropvalOutResponse> {
+    const queryString = toURLSearchParams({
+      dcids: params.dcids,
+      prop: params.prop,
+    });
+    const url = `${this.apiRoot || ""}/api/node/propvals/out?${queryString}`;
+    return (await (await fetch(url)).json()) as ApiNodePropvalOutResponse;
```

Comment:
nit: maybe split the awaits to two separate lines for easier reading

Reply:
Done (fixed other instances as well)